### PR TITLE
fix: resource updater + updated data

### PR DESCRIPTION
**Pull request by Constrat**
ref: 
- #7074

### CODE REVIEW

**Comment by Constrat** [comment]
Why does the resource updater change the order of the tasks.json: https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/7079/commits/939aaaed0bf15b57d68175b43e6053b86b52175c

**Comment by Constrat** [comment]
I've tried everything to remove the alphabetic ordering but I can't seem to find what's the cause. My only idea that I have left is that C++ doesn't like "overwriting" the opened file but that doesn't make much sense.
Apart from this small issue the fix is done.

As I said in #7079 the Taiwanese server won't have any more updates as I couldn't find any way to extract the in-game data so for now I've just commented it out
```cpp
     std::unordered_map<std::string, std::string> global_dirs = {
        { "en_US", "YoStarEN" }, { "ja_JP", "YoStarJP" }, { "kvo_KR", "YoStarKR" },
        //{ "tw", "txwy" },
    };
```
and removed it from the operator localization algorithm (It will display the zh_CN name, as I couldn't leave it empty `""`. Removing the `"name_tw"` field will break everything inside both MaaCore and MaaWpfGui)

**Comment by MistEO** [comment]
> Why does the resource updater change the order of the tasks.json: [939aaae](https://github.com/MaaAssistantArknights/MaaAssistantArknights/commit/939aaaed0bf15b57d68175b43e6053b86b52175c)
> 
> ```c++
>     std::ofstream ofs(tasks_path, std::ios::out);
>     ofs << task_json.format() << std::endl;
> ```
> 
> This somehow orders the json alphabetically, which we decided is stupid and doesn't make sense. I understand that running res_update_game.yml will re order them with the python script, but I feel like it's just wasted computation

It is not sorted. According to the JSON standard, json objects are out of order, which depends on the implementation of the json library, and our implementation is alphabetical order.

**Review by MistEO — APPROVED** [review]
It's a shame that TXWY will no longer get updates, but there's nothing we can do.

**Comment by MistEO** [comment]
We can read the name_tw field in the previous battle_data.json. If it exists, use the previous one. If not, use the name field, this can save some names.
And if someone wants to manually update the name_tw of the new operator, it will not be overwritten by the automatic updater.

**Comment by Constrat** [comment]
> We can read the name_tw field in the previous battle_data.json. If it exists, use the previous one. If not, use the name field, this can save some names. And if someone wants to manually update the name_tw of the new operator, it will not be overwritten by the automatic updater.

Then we can use this method for all the files not only battle_chars.
We would just read the latest existing version, and all the modifications are going to be made via manual commit

**Comment by Constrat** [comment]
Wait for new TW fix before merge

**Comment by Constrat** [comment]
@MistEO With MeoJson how can I use 
```hpp
MEOJSON_INLINE std::optional<value_t> basic_value<string_t>::find(const string_t& key) const
{
    return is_object() ? as_object().template find<value_t>(key) : std::nullopt;
}
```
To search the id of the characters in this case? `"chars" / id`?
```json
{
    "chars": {
        "char_002_amiya": {
            "name": "阿米娅",
            "name_en": "Amiya",
            "name_jp": "アーミヤ",
            "name_kr": "아미야",
            "name_tw": "阿米婭",
            "position": "RANGED",
            "profession": "CASTER",
            "rangeId": [
                "3-6",
                "3-1",
                "3-1"
            ],
            "rarity": 5
        },

```

**Comment by Constrat** [comment]
I FINALLY DID IT
@MistEO ... This is a suggestion but... We need to work on the documentation xD.

**Comment by Constrat** [comment]
I'm finished 😅😅😅
Considering the amount of commits, squashing is definitely the best choice

File: tools/ResourceUpdater/main.cpp
Comment (MistEO) [inline tools/ResourceUpdater/main.cpp:844]: try `char_new_data["name_tw"] = battle_data_json.get("chars", id, cn_default_name);`

**Comment by MistEO** [comment]
It's late at night here, I'm going to sleep xD, and I didn't check carefully. You can merge it and release a nightly version, overall, there is no big problem.

File: tools/ResourceUpdater/main.cpp
Comment (Constrat) [reply tools/ResourceUpdater/main.cpp:844]: Is it that easy and compact? XD

Comment (Constrat) [reply tools/ResourceUpdater/main.cpp:844]: Ok this is game changing. I can implement `.get` in the operator localization that I implemented a week ago. tyvm

**Comment by Constrat** [comment]
```cpp
        for (int i = 1; i < chars_json.size(); i++) {
            char_new_data[chars_json[i].second] =
                (i != 4) ? chars_json[i].first.get(id, "name", default_name)
                         : chars_json[i].first.get("chars", id, chars_json[i].second, default_name);
        }

```
It's beatiful

**Comment by MistEO** [comment]
chars_json[i].first.get(id, "name", default_name)
->
chars_json.get(i, 0, id, "name", default_name)
😎

**Comment by Constrat** [comment]
> chars_json[i].first.get(id, "name", default_name) -> chars_json.get(i, 0, id, "name", default_name) 😎

Oh cool, but since I could squash all the clients in the same algorithm I chose a foreach (looks better)

**Comment by AnnAngela** [comment]
@Constrat Hi, I got a data downloader from arkntools to download the data for TXWY Server version, please let me know if you need. Here is the list of the tables of the data:
- character_table
- char_patch_table
- building_data
- skill_table
- gacha_table
- item_table
- stage_table
- zone_table
- gamedata_const
- activity_table
- retro_table
- uniequip_table

**Comment by AnnAngela** [comment]
Me too \_(:з」∠)_ Too busy to look at it

**Comment by Constrat** [comment]
I'll see what I can do this week/weekend

**Pull request merged**
